presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/api/ControllerControlAPI.java

```diff
@@ -0,0 +1,114 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.teradata.prestomanager.controller.api;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.teradata.prestomanager.common.ApiRequester;
+import com.teradata.prestomanager.common.StopType;
+import com.teradata.prestomanager.controller.AbstractControllerAPI;
+import com.teradata.prestomanager.controller.RequestDispatcher;
+
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import java.util.List;
+import java.util.UUID;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static org.eclipse.jetty.http.HttpMethod.GET;
+import static org.eclipse.jetty.http.HttpMethod.POST;
+
+@Path("/presto")
+@Singleton
+public class ControllerControlAPI
+        extends AbstractControllerAPI
+{
+    @Inject
+    public ControllerControlAPI(
+            Client forwardingClient,
+            RequestDispatcher requestDispatcher)
+    {
+        super(forwardingClient, requestDispatcher);
+    }
+
+    @POST
+    @Path("/start")
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response startPresto(
+            @QueryParam("scope") String scope,
+            @QueryParam("nodeId") List<UUID> nodeId)
+    {
+        ApiRequester apiRequester = requesterBuilder(ControllerControlAPI.class)
+                .httpMethod(POST)
+                .accept(MediaType.TEXT_PLAIN)
+                .pathMethod("startPresto")
+                .build();
+
+        return forwardRequest(scope, apiRequester, nodeId);
+    }
+
+    @POST
+    @Path("/stop")
+    public Response stopPresto(
+            @QueryParam("stopType") @DefaultValue("GRACEFUL") StopType stopType,
+            @QueryParam("scope") String scope,
+            @QueryParam("nodeId") List<UUID> nodeId)
+    {
+        ApiRequester.Builder apiRequester = requesterBuilder(ControllerControlAPI.class)
+                .httpMethod(POST)
+                .pathMethod("stopPresto");
+
+        OptionalQueryParam(apiRequester, "stopType", stopType);
+
+        return forwardRequest(scope, apiRequester.build(), nodeId);
+    }
+
+    @POST
+    @Path("/restart")
+    public Response restartPresto(
+            @QueryParam("scope") String scope,
+            @QueryParam("nodeId") List<UUID> nodeId)
+    {
+        ApiRequester apiRequester = requesterBuilder(ControllerControlAPI.class)
```

Comment:
What type of stop does restart do? It seems like (except for the coordinator) you could choose to have a graceful stop here too.